src/credentials/attestation_verifier/JSONFileRevocationSet.cpp

```diff
@@ -0,0 +1,189 @@
+#include <lib/support/BytesToHex.h>
+#include <lib/support/ScopedBuffer.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+
+namespace chip {
+namespace Credentials {
+
+JSONFileRevocationSet::JSONFileRevocationSet(const char * revocationSetPath)
+{
+    VerifyOrReturn(revocationSetPath != nullptr);
+
+    if (revocationSetPath != nullptr)
+    {
+        LoadJSONRevocationSet(revocationSetPath, mRevocationSet);
+        VerifyOrReturn(revocationSetCount());
+    }
+
+    mIsInitialized = true;
+}
+
+void LoadJSONRevocationSet(const char * revocationSetPath, Json::Value & revocationSet)
+{
+    VerifyOrReturn(revocationSetPath != nullptr);
+
+    std::ifstream ifs(revocationSetPath);
+    Json::Reader reader;
+    reader.parse(ifs, revocationSet);
+}
+
+AttestationVerificationResult JSONFileRevocationSet::IsCertificateRevoked(bool isPaa, Crypto::AttestationCertVidPid vidPidUnderTest,
+                                                                          ByteSpan issuer, ByteSpan authorityKeyId,
+                                                                          ByteSpan serialNumber) const
+{
+    for (size_t revocation_set_idx = 0; revocation_set_idx < mRevocationSet.size(); ++revocation_set_idx)
+    {
+        Json::Value type = mRevocationSet[static_cast<int>(revocation_set_idx)]["type"];
+
+        // 1.
+        if (type.asString().compare("revocation_set") == 0)
+        {
+            Json::Value jsonCrlIssuerSubjectKeyId = mRevocationSet[static_cast<int>(revocation_set_idx)]["issuer_subject_key_id"];
+            Json::Value jsonCrlIssuerName         = mRevocationSet[static_cast<int>(revocation_set_idx)]["issuer_name"];
+            Json::Value jsonCrlRevokedSerialNumbers =
+                mRevocationSet[static_cast<int>(revocation_set_idx)]["revoked_serial_numbers"];
+
+            uint8_t crlIssuerSubjectKeyIdBuf[Crypto::kAuthorityKeyIdentifierLength] = { 0 };
+            ByteSpan crlIssuerSubjectKeyId(crlIssuerSubjectKeyIdBuf);
+
+            VerifyOrReturnError(Encoding::HexToBytes(jsonCrlIssuerSubjectKeyId.asString().c_str(),
+                                                     jsonCrlIssuerSubjectKeyId.asString().size(), crlIssuerSubjectKeyIdBuf,
+                                                     sizeof(crlIssuerSubjectKeyIdBuf)) == Crypto::kAuthorityKeyIdentifierLength,
+                                AttestationVerificationResult::kInvalidArgument);
+
+            size_t crlSignerCertificateMaxLength = BASE64_MAX_DECODED_LEN(jsonCrlIssuerName.asString().size());
+
+            Platform::ScopedMemoryBuffer<uint8_t> crlSignerCertificate;
+
+            ReturnErrorCodeIf(!crlSignerCertificate.Alloc(crlSignerCertificateMaxLength), AttestationVerificationResult::kNoMemory);
+
+            // 2.
+            size_t crlSignerCertificateLength =
+                Base64Decode(jsonCrlIssuerName.asString().c_str(), jsonCrlIssuerName.asString().size(), crlSignerCertificate.Get());
```

Comment:
What guarantees this is not a buffer overrun?

Reply:
Because the crlSignerCertificate buffer was allocated with size from the macro BASE64_MAX_DECODED_LEN just a few lines above.

Reply:
Ah, I see.  OK.